src/coreclr/tools/aot/ILCompiler.Compiler.Tests/ILCompiler.Compiler.Tests.Assets/SwiftTypes.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace ILCompiler.Compiler.Tests.Assets.SwiftTypes;
+
+[ExpectedLowering(ExpectedLoweringAttribute.Lowered.Int64, ExpectedLoweringAttribute.Lowered.Double, ExpectedLoweringAttribute.Lowered.Int64, ExpectedLoweringAttribute.Lowered.Int16)]
+public struct I64_D_I8_I32_UI16
+{
+    public long i64;
+    public double d;
+    public sbyte i8;
+    public int i32;
+    public ushort ui16;
+}
+
+[ExpectedLowering(ExpectedLoweringAttribute.Lowered.Int64, ExpectedLoweringAttribute.Lowered.Double, ExpectedLoweringAttribute.Lowered.Int64, ExpectedLoweringAttribute.Lowered.Int8)]
+public struct I64_D_I8_I32_UI8
+{
+    public long i64;
+    public double d;
+    public sbyte i8;
+    public int i32;
+    public byte u8;
+}
+
+[ExpectedLowering(ExpectedLoweringAttribute.Lowered.Int16)]
+public struct F5_S1_S0
+{
+    public short F0;
+}
+
+[ExpectedLowering(ExpectedLoweringAttribute.Lowered.Int64, ExpectedLoweringAttribute.Lowered.Int64, ExpectedLoweringAttribute.Lowered.Int16, ExpectedLoweringAttribute.Lowered.Int64)]
+public struct F5_S1
+{
+    public ulong F0;
+    public long F1;
+    public F5_S1_S0 F2;
+    public long F3;
+}
+
+[ExpectedLowering(ExpectedLoweringAttribute.Lowered.Int32)]
+public struct F5_S2_S0
+{
+    public short F0;
+    public byte F1;
+}
+
+[ExpectedLowering(ExpectedLoweringAttribute.Lowered.Int64, ExpectedLoweringAttribute.Lowered.Int64, ExpectedLoweringAttribute.Lowered.Int32, ExpectedLoweringAttribute.Lowered.Int64)]
+public struct F5_S2
+{
+    public ulong F0;
+    public long F1;
+    public F5_S2_S0 F2;
+    public long F3;
+}
```

Comment:
I assume the last `Int64` here starts at offset 24, yet the primitives before it sum only to 20 in size. What is the right way for the JIT to figure out what part of the struct should corresponds to which specific primitive in the sequence?
It seems like the JIT-EE API should also include offsets for each primitive.

Reply:
The JIT should assume natural alignment for each element in the primitive sequence. Any empty space in the middle of the struct due to alignment is unused padding in the struct that can be ignored.

That's how the generated code seems to look for the cases I've inspected.